### PR TITLE
Feature/carousel on class

### PR DESCRIPTION
**Pull request by d-tamirlan**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 0.5.3 (Aug 3, 2021)
```
Comment (rrrrs09) [inline CHANGELOG.md:1]:
```suggestion
## 0.5.0 (Aug 3, 2021)
```

File: pybotx_widgets/base.py [hunk 2]
```diff
@@ -4,11 +4,6 @@
 
 
 class Widget:
```
Comment (rrrrs09) [inline pybotx_widgets/base.py:6]: Maybe you should rename it to `WidgetProtocol`?

**Review by YunusovSamat — APPROVED** [review]

File: pybotx_widgets/calendar.py [hunk 3]
```diff
@@ -277,3 +264,10 @@ async def display(self) -> Optional[date]:
         await send_or_update_message(
             self.message, self.bot, self.SELECT_DATE, self.markup
         )
+
+
+def _clear_calendar_data(message: Message) -> None:
+    """Clear widget data form message.data."""
```
Comment (rrrrs09) [inline pybotx_widgets/calendar.py:270]: Did you mean `from`?

File: pybotx_widgets/carousel.py [hunk 4]
```diff
@@ -1,317 +1,324 @@
...
-
-def _clear_carousel_data(message: Message) -> None:
-    """Clear widget data form message.data."""
-
-    message.data.pop("message_id", None)
-    message.data.pop("carousel_selected_val", None)
-    message.data.pop("carousel_selected_value_label", None)
-    message.data.pop("carousel_message_label", None)
-    message.data.pop("carousel_start_from", None)
+        if len([place for place in number_places if place[1] == ""]) != 2:
```
Comment (rrrrs09) [inline pybotx_widgets/carousel.py:37]: Don't you think this expression is hard to read? Maybe move it above?

File: pybotx_widgets/carousel.py [hunk 5]
```diff
@@ -1,317 +1,324 @@
...
-    show_left_arrow: bool,
-    show_right_arrow: bool,
-    show_numbers: bool,
-    pagination: PaginationInfo,
-    control_labels: Tuple[str, str],
-    additional_markup: MessageMarkup = None,
-) -> MessageMarkup:
-    """Build newline markup for Carousel widget."""
+        if not self.control_labels:
+            self.control_labels = (
```
Comment (rrrrs09) [inline pybotx_widgets/carousel.py:56]: Nit: maybe use simple `if` statement?

File: pybotx_widgets/carousel.py [hunk 6]
```diff
@@ -1,317 +1,324 @@
...
+                (f"{self.LEFT_ARROW} ({{}}-{{}})", f"{self.RIGHT_ARROW} ({{}}-{{}})")
+                if self.show_numbers
+                else (self.LEFT_ARROW, self.RIGHT_ARROW)
+            )
 
-    markup = MessageMarkup()
-    right_label = control_labels[1]
+        if self.show_numbers:
+            if not self._is_valid_label_for_placing_numbers(self.control_labels[0]):
```
Comment (rrrrs09) [inline pybotx_widgets/carousel.py:63]: Nit: you can use properties for left and right control labels.

File: pyproject.toml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pybotx-widgets"
-version = "0.4.3"
+version = "0.5.3"
```
Comment (rrrrs09) [inline pyproject.toml:3]:
```suggestion
version = "0.5.0"
```

File: pybotx_widgets/calendar.py (hunk 3 above)
Comment (d-tamirlan) [reply pybotx_widgets/calendar.py:270]: yes

File: CHANGELOG.md (hunk 1 above)
Comment (d-tamirlan) [reply CHANGELOG.md]: done

File: pybotx_widgets/calendar.py (hunk 3 above)
Comment (d-tamirlan) [reply pybotx_widgets/calendar.py:270]: done

File: pybotx_widgets/carousel.py (hunk 4 above)
Comment (d-tamirlan) [reply pybotx_widgets/carousel.py]: remove that expression

File: pybotx_widgets/carousel.py (hunk 5 above)
Comment (d-tamirlan) [reply pybotx_widgets/carousel.py]: done

File: pybotx_widgets/carousel.py (hunk 6 above)
Comment (d-tamirlan) [reply pybotx_widgets/carousel.py]: done

File: pyproject.toml (hunk 7 above)
Comment (d-tamirlan) [reply pyproject.toml]: done

File: pybotx_widgets/carousel.py [hunk 8]
```diff
@@ -280,6 +288,7 @@ def displayed_content(self) -> Iterator:
 
     @classmethod
     async def get_value(cls, message: Message, bot: Bot) -> Optional[str]:
+        print('message.data', message.data)
```
Comment (rrrrs09) [inline pybotx_widgets/carousel.py:291]: Forgot

**Review by rrrrs09 — APPROVED** [review]

File: pybotx_widgets/carousel.py (hunk 8 above)
Comment (d-tamirlan) [reply pybotx_widgets/carousel.py]: Oops

**Pull request merged**